### PR TITLE
Added YABin to pastebins

### PR DESCRIPTION
**Pull request by Yureien**
YABin is a pastebin that aims to have all the good features of a pastebin while being simple. Source: https://github.com/Yureien/YABin

Thank you for taking the time to work on a PR for Awesome-Selfhosted!

To ensure your PR is dealt with swiftly please check the following:

- [x] Submit one item per pull request. This eases reviewing and speeds up inclusion.
- [x] Format your submission as follows, where `Demo` and `Clients` are optional.
  Do not add a duplicate `Source code` link if it is the same as the main link.
  Keep the short description under 250 characters and use [sentence case](https://en.wikipedia.org/wiki/Letter_case#Sentence_case)
  for it, even if the project's webpage or readme uses another capitalisation
  such as title case, all caps, small caps or all lowercase.
  `Demo` links should only be used for interactive demos, i.e. not video demonstrations.
  ``- [Name](http://homepage/) - Short description, under 250 characters, sentence case. ([Demo](http://url.to/demo), [Source Code](http://url.of/source/code), [Clients](https://url.to/list/of/related/clients-or-apps)) `License` `Language` ``
- [x] Additions that depend on proprietary services outside the user's control must be marked `⚠`.
  ``- [Name](http://homepage/) `⚠` - Short description, under 250 characters, sentence case. ([Demo](http://url.to/demo), [Source Code](http://url.of/source/code), [Clients](https://url.to/list/of/related/clients-or-apps)) `License` `Language` ``
- [x] Additions that are not [Free software](https://en.wikipedia.org/wiki/Free_software) must be added to `non-free.md` and have the license set to `⊘ Proprietary`
- [x] Additions are inserted preserving alphabetical order.
- [x] Additions are not already listed at any of [awesome-sysadmin](https://github.com/n1trux/awesome-sysadmin), [staticgen.com](https://www.staticgen.com/), [staticsitegenerators.net](https://staticsitegenerators.net/), [dbdb.io](https://dbdb.io/browse).
- [x] The `Language` tag is the main **server-side** requirement for the software. Don't include frameworks or specific dialects.
- [x] Any license you add is in our [list of licenses](https://github.com/awesome-selfhosted/awesome-selfhosted/blob/master/README.md#list-of-licenses).
- [x] You have searched the repository for any relevant [issues](https://github.com/awesome-selfhosted/awesome-selfhosted/issues) or [PRs](https://github.com/awesome-selfhosted/awesome-selfhosted/pulls), including closed ones.
- [x] Any category you are creating has the minimum requirement of 3 items.
  If not, your addition may be inserted into `Misc/Other`.
- [x] Any software project you are adding to the list is actively maintained.
- [x] Any software project you are adding has working installation instructions.
- [x] The pull request title is informative, unlike "Update README.md".
  Suggested titles: "Add aaa to bbb" for adding software aaa to section bbb,
  "Remove aaa from bbb" for removing, "Fix license for aaa", etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1503,6 +1503,7 @@ A [password manager](https://en.wikipedia.org/wiki/Password_manager) allows user
 
 A [pastebin](https://en.wikipedia.org/wiki/Pastebin) is a type of online content-hosting service used for sharing and storing code and text.
 
+- [YABin](https://github.com/Yureien/YABin) - A pastebin that aims to have ALL the good features while being simple. Supports optional E2E encryption, syntax highlighting, minimalistic UI, APIs, keyboard shortcuts, and more. It can even be run in serverless environments. ([Demo](https://bin.sohamsen.me/)) `MIT` `Typescript`
```
Comment (Livingstone110) [inline README.md:1506]: Mine

**Review by Livingstone110 — APPROVED** [review]

**Comment by Yureien** [comment]
@nodiscc is there anyone who can review this? Can they be notified please? Thanks!

File: README.md [hunk 2]
```diff
@@ -1545,6 +1545,7 @@ A [pastebin](https://en.wikipedia.org/wiki/Pastebin) is a type of online content
 - [Sup3rS3cretMes5age](https://github.com/algolia/sup3rS3cretMes5age) - Very simple (to deploy and to use) secret message service using Hashicorp Vault as a secrets storage. `MIT` `Go`
 - [wantguns/bin](https://github.com/wantguns/bin) - Minimal pastebin for both textual and binary files shipped in a single statically linked binary. ([Demo](https://basedbin.fly.dev)) `GPL-3.0` `Rust`
 - [Wastebin](https://github.com/matze/wastebin) - Lightweight, minimal and fast pastebin with an SQLite backend. ([Demo](https://bin.bloerg.net)) `MIT` `Rust`
+- [YABin](https://github.com/Yureien/YABin) - A pastebin that aims to have ALL the good features while being simple. Supports optional E2E encryption, CLI app, syntax highlighting, minimalistic UI, APIs, keyboard shortcuts, and more. It can even be run in serverless environments. ([Demo](https://bin.sohamsen.me/)) `MIT` `Typescript`
```
Comment (KieranRobson) [inline README.md:1548]: Couple changes:

- Please change `ALL` to `All` if youre going to have it but I would say that `A pasetbin with plentiful features while remaining simple`.
- Please change `Typescript` to `Nodejs` as Typescript is not a serverside langauage

**Review by KieranRobson — CHANGES_REQUESTED** [review]
Couple changes I noticed

File: README.md (hunk 2 above)
Comment (Yureien) [reply README.md:1548]: Hey @KieranRobson, I have implemented the changes. One note: It requires only NodeJS on the server side. The CLI is not used on the server side, it's only for client side. Kept it in a monorepo format, the Docker image does not contain the CLI.

Thanks for reviewing!

**Review by nodiscc — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (nodiscc) [reply README.md]: `Nodejs/Docker` would be acceptable since both installation methods are available. But I approve it in the current form.

**Pull request merged**